.circleci/config.yml

```diff
@@ -213,6 +201,24 @@ jobs:
           name: Run memory tests
           command: postgrest-test-memory
           when: always
+      - run:
+          name: Run load test
+          command: postgrest-loadtest
+          when: always
+      - run:
+          name: Build all derivations from default.nix
+          command: nix-build
```

Comment:
The `nix-build` is now run after all the tests in the pipeline. How are the tests now run against the new build? It looks like they are run against the cached version from cachix, which would always be the last build from master? What am I missing?

Reply:
Good point @wolfgangwalther , I'll need to add a comment to document this. This setup works correctly and makes the tests run a bit earlier in the job, failing faster if needed.

`nix-env -iA` in the earlier step also uses `nix-build` under the hood. At this point, only the things that are really needed to run the tests are built or pulled from the cache if available. `nix-env -iA` will never install anything stale from the cache. 

This `nix-build` call at  this step then just builds whatever is not yet built or pulled from the cache.

Reply:
Ah cool. Just had a comment on that previous step, that it should maybe be renamend then. Deleted it again, after I realized there are actually 2 steps that have `nix-env -iA` in them. So a comment + rename of one of the steps would sure help, I guess.

I understand that we could basically run the tests and `nix-build` in parallel after we are done with `Install testing scripts`? Have you tried doing that with circleCI's "workspaces" feature?
It looks like you could split this into 3 jobs, where the test and build jobs run in parallel after the first one. I guess you could just copy the whole `/nix` folder between those jobs via persist_to_workspace / attach_workspace.

See here for reference:
https://circleci.com/blog/persisting-data-in-workflows-when-to-use-caching-artifacts-and-workspaces/
https://circleci.com/blog/deep-diving-into-circleci-workspaces/

One benefit of that would be that those jobs would both show up separately in the check list on github. I think this could then be extended to all the tests against different pg versions - to have them show up one by one on the check list. Only if that's wanted, of course.

Reply:
Using workspaces is a cool idea! We'll need to test how this performs, as the `/nix/` folder can easily be a few GB large, having a few GHC versions for the static builds etc. Not sure if 'saving' and 'loading' the workspace could be a bottleneck (it would be with the CircleCI caching feature, but maybe workspaces work differently).

Reply:
> as the `/nix/` folder can easily be a few GB large, having a few GHC versions for the static builds etc

Shouldn't those be fetched only at the `nix-build` stage?

But yeah performance needs to be tested. Very likely that it won't actually perform better.